src/prefect/agent/agent.py

```diff
@@ -286,13 +286,14 @@ def _log_flow_run_exceptions(self, flow_runs: list, exc: Exception) -> None:
             - flow_runs (list): A list of GraphQLResult flow run objects
             - exc (Exception): A caught exception to log
         """
-        for flow_run in flow_runs:
-            self.client.write_run_log(
-                flow_run_id=flow_run.id,  # type: ignore
-                name="agent",
-                message=str(exc),
-                level="ERROR",
-            )
+        if flow_runs:
```

Comment:
Sorry for the nitpick here, but I think this might be cleaner if we instead updated line 154 with:
```python
self._log_flow_run_exceptions(flow_runs or [], exc)  # type: ignore
```
and still kept the iteration inside the log flow run piece.